modules/oadp-using-enable-api-group-versions.adoc

```diff
@@ -0,0 +1,49 @@
+// Module included in the following assemblies:
+//
+// * backup_and_restore/application_backup_and_restore/advanced-topics.adoc
+
+
+:_content-type: PROCEDURE
+[id="oadp-using-enable-api-group-versions_{context}"]
+= Working with different Kubernetes API versions in the same OADP operation
+
+By default, Velero only backs up resources that use the preferred version of the Kubernetes API. However, Velero also includes a feature, link:https://velero.io/docs/v1.9/enable-api-group-versions-feature/[Enable API Group Versions], that overcomes this limitation. When enabled on the source cluster, this feature causes Velero to back up _all_ Kubernetes API group versions that are supported on the cluster, not only the preferred one. Once the versions are stored in the backup tarball file, they are available to be restored on the destination cluster.
```

Comment:
@kaovilai, I thought API objects tended to be upgraded automatically to the latest version; A cluster can have API objects that are against an older API persistently?

I'm just trying to better understand when this situation comes into play.

Thanks!

Reply:
It's not automatic. If your yaml uses a specific version, it will use that version and fail if it doesn't exist in the cluster. Cluster can maintain multiple valid versions at any given point in time.

Reply:
For kube native or some controllers it's possible they do upgrade behind the scenes.. but if you oc create v1alpha1 I wouldn't expect everything with v1 to get upgraded to v1.

There could've been many changes between the two and automatic upgrade on everything would be problematic.